source/aws_signing.c

```diff
@@ -1648,6 +1626,55 @@ int aws_signing_build_authorization_value(struct aws_signing_state_aws *state) {
         goto cleanup;
     }
 
+    /*
+     * Add X-Amz-Date to the signing result
+     */
+    struct aws_byte_cursor date_header_name = aws_byte_cursor_from_string(g_aws_signing_date_name);
+    struct aws_byte_cursor date_header_value = aws_byte_cursor_from_buf(&state->date);
+    if (aws_signing_result_append_property_list(
+            state->result, g_aws_http_headers_property_list_name, &date_header_name, &date_header_value)) {
+        return AWS_OP_ERR;
+    }
+
+    /*
+     * Add Security token to the signing result if a session token was present.
+     */
+    if (state->config->credentials->session_token) {
+        struct aws_byte_cursor session_token_name = aws_byte_cursor_from_string(g_aws_signing_security_token_name);
+        struct aws_byte_cursor session_token = aws_byte_cursor_from_string(state->config->credentials->session_token);
+
+        const struct aws_string *property_list_name = g_aws_http_headers_property_list_name;
+
+        struct aws_byte_buf uri_encoded_buf;
+        AWS_ZERO_STRUCT(uri_encoded_buf);
+
+        int error_code = AWS_OP_ERR;
+        if (s_is_query_param_auth(state->config->algorithm)) {
```

Comment:
Can you just add a comment about what this is doing. I barely understand it as I read it. The rest makes sense